### ISSUE TITLE
False positive ``not-callable`` for ``typing.NamedTuple``

### ISSUE THREAD

**Issue by meuter** [PyCQA/pylint#4431]
### Steps to reproduce

Given a file `bug.py`:
```python
from typing import NamedTuple

Bar = NamedTuple("Bar", foo=int, baz=str)     
Bar(10, "yo")
```

### Current behavior

Result of `pylint bug.py`:
```
************* Module bug
bug.py:4:0: E1102: Bar is not callable (not-callable)

--------------------------------------------------------------------
Your code has been rated at -6.67/10 (previous run: -6.67/10, +0.00)
```

### Expected behavior

Pylint should not complain as `Bar` is callable

### pylint --version output

Result of `pylint --version` output:

```
pylint 2.8.2
astroid 2.5.6
Python 3.9.4 (tags/v3.9.4:1f2e308, Apr  6 2021, 13:40:21) [MSC v.1928 64 bit (AMD64)]
```

OS: Windows 10
Shell: PowerShell 7

**Comment by timmartin** [comment]
It looks to me like this is failing to be inferred in Astroid, [here in `infer_typing_namedtuple`](https://github.com/PyCQA/astroid/blob/8f70c7dade73817a19b686bbb463b1aba0c7397a/astroid/brain/brain_namedtuple_enum.py#L493). Astroid is expecting there to be exactly two arguments to the namedtuple, and your code has one (plus kwargs).

Looking at the docs for `typing.NamedTuple`, it's not clear to me that it's supposed to work the way you're calling it. The example given in the docs is:

    Employee = NamedTuple('Employee', [('name', str), ('id', int)])

i.e. with exactly two args, and if you write it this way it seems to not cause a `not-callable` error from Pylint.

Is the alternative form of calling `NamedTuple` with kwargs documented somewhere? If it's a recent addition, I guess Astroid is just out of date.

**Comment by Pierre-Sassoulas** [comment]
Closing as I can't reproduce on 2.15 so it's probably been fixed.
